git_machete/utils.py

```diff
@@ -257,3 +193,90 @@ def slurp_file_or_empty(path: str) -> str:
             return file.read()
     except IOError:
         return ''
+
+
+class AnsiEscapeCodes:
+    stdout = popen_cmd('tput', 'colors')[1]
+    try:
+        __number_of_supported_colors = int(stdout)
+    except ValueError:
+        # If we cannot retrieve the number of supported colors, let's defensively assume it's low.
+        __number_of_supported_colors = 8
+
+    # `GIT_MACHETE_DIM_AS_GRAY` remains undocumented as for now,
+    # was just needed for animated gifs to render correctly (`[2m`-style dimmed text was invisible)
+    __dim_as_gray = os.environ.get('GIT_MACHETE_DIM_AS_GRAY') == 'true'
+
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    DIM = '\033[38;2;128;128;128m' if __dim_as_gray else '\033[2m'
+    # Let's fall back to cyan on 8-color terminals
+    UNDERLINE = '\033[4m' if __number_of_supported_colors >= 256 else '\033[36m'
+    GREEN = '\033[32m'
+    YELLOW = '\033[33m'
+    # Let's fall back to yellow on 8-color terminals
+    ORANGE = '\033[00;38;5;208m' if __number_of_supported_colors >= 256 else '\033[33m'
+    # Let's fall back to dark red (which might be similar to yellow :/) on 8-color terminals
+    RED = '\033[91m' if __number_of_supported_colors >= 256 else '\033[31m'
```

Comment:
`__number_of_supported_colors >= 256` consider extracting that boolean to a variable